bin.js

```diff
@@ -50,16 +50,29 @@ function degit(dir) {
 
 async function patch(dir) {
   try {
-    if (!fs.existsSync(join(dir, 'src/server.js'))) {
+    const srcDir = fs.readdirSync(join(dir, 'src'))
+    const serverFile = srcDir.find((file) => file.includes('server'))
+
+    if (!serverFile) {
       throw new Error(
-        'expected src/server.js to exist, is this really a Sapper project?'
+        `Expected src/server.{js|ts} to exist, is this really a Sapper project?`
       )
     }
-    const server = fs.readFileSync(join(dir, 'src/server.js'), 'utf8')
-    fs.writeFileSync(join(dir, 'src/server.js'), patchServer(server))
-    console.log(green('Patched'), 'src/server.js')
+
+    const server = fs.readFileSync(join(dir, `src/${serverFile}`), 'utf8')
+    fs.writeFileSync(
+      join(dir, `src/${serverFile}`),
+      patchServer(server, serverFile)
+    )
+    console.log(green('Patched'), `src/${serverFile}`)
   } catch (e) {
-    console.error(red(e.message))
+    console.error(
+      red(
+        e.code === 'ENOENT'
+          ? 'No src folder found, is this really a Sapper project?'
```

Comment:
In what case could this error be thrown?

Reply:
`fs.readdirSync` leads to `ENOENT` when the requested directory doesn't exist.

I implemented your suggested solution with checking for `fs.existsSync` twice, so I changed this part back to how it was before.